python/pyarrow/tests/test_table.py

```diff
@@ -1989,6 +1989,7 @@ def test_table_select():
     assert result.equals(expected)
 
 
+@pytest.mark.acero
 def test_table_group_by():
```

Comment:
There are some other `join` tests in this file marked with `pytest.mark.dataset` that can then be changed to "acero"

Reply:
In theory we could, but getting the checks like `not isinstance(left_operand, (Table, ds.Dataset))` to work without datasets sounds like it's more trouble than it's worth at the moment.

Reply:
Actually found a way to do that, writing a little stub class for the module. LMK if there's a better way we would prefer to handle that.